internal/config/config.go

```diff
@@ -851,26 +852,34 @@ func loadConfig(config string) ([]byte, error) {
 }
 
 func fetchConfig(u *url.URL) ([]byte, error) {
-	req, err := http.NewRequest("GET", u.String(), nil)
+	// Read the startup flag for config-http-retry and try to convert to integer
+	configHttpRetry, err := strconv.Atoi(flag.Lookup("config-http-retry").Value.String())
 	if err != nil {
-		return nil, err
-	}
-
-	if v, exists := os.LookupEnv("INFLUX_TOKEN"); exists {
-		req.Header.Add("Authorization", "Token "+v)
-	}
-	req.Header.Add("Accept", "application/toml")
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to process config-http-retry ")
 	}
+	// Loop to retry http config request to prevent agent dying on startup if endpoint is unavailable
+	for {
```

Comment:
should probably support max-retries before permanently failing. Not a big fan of looping forever.

Reply:
Sure, I did propose a CLI flag to set max retries further up the chain, but understand it adds to the mass of options.

Personally, I would like it to return forever, otherwise someone it means some manual intervention is required, but others may not so think an option would be useful to drive desired behaviour.

Would using an environment variable setting be an alternative approach to a CLI switch here?
e.g. export TELEGRAF_HTTP_RETRIES=50
It not set, then we just use a default of (say) 10 iterations every 1 minute.

If so, then I'll take a look at building this in an resubmitting for review.